File: golang-be/Tours/model/TourDuration.go
package model

type TransportationType int

type TourDuration struct {
	Id             int                `bson:"_id"` // If you want to use a custom ID, use "_id" field name
	TourID         int                `bson:"tour_id"`
	TimeInSeconds  int                `bson:"time_in_seconds"`
	Transportation TransportationType `bson:"transportation"` // Assuming TransportationType is a string enum
}

func NewTourDuration(timeInSeconds int, transpotation TransportationType) TourDuration {
	return TourDuration{TimeInSeconds: timeInSeconds, Transportation: transpotation}
}

const (
	Walking TransportationType = iota
	Bicycle
	Car
)

File: tourism-be/src/Modules/Tours/Explorer.Tours.Core/Domain/Tours/TourDuration.cs
using Explorer.BuildingBlocks.Core.Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Explorer.Tours.Core.Domain.Tours
{
    public class TourDuration : ValueObject
    {
        public uint TimeInSeconds { get; init; }
        public TransportationType Transportation { get; init; }

        [JsonConstructor]
        public TourDuration(uint timeInSeconds, TransportationType transportation)
        {
            TimeInSeconds = timeInSeconds;
            Transportation = transportation;
        }
        protected override IEnumerable<object> GetEqualityComponents()
        {
            yield return TimeInSeconds;
            yield return Transportation;
        }
    }
    public enum TransportationType
    {
        Walking,
        Bicycle,
        Car
    }
}
